pages/Events.tsx

```diff
@@ -0,0 +1,111 @@
+// Style
+import styles from '../styles/Events.module.scss';
+// Internal Components
+import NavBar from '../components/NavBar';
+import Footer from '../components/Footer';
+import Section from '../layouts/Section';
+import EventBanner from '../components/EventBanner';
+import Event from '../components/Event';
+import {
+  events,
+  SectionType,
+  EventGridStyle,
+  type EventItem,
+  type EventSection,
+} from '../config';
+
+const processEvents = (events: EventItem[]) => {
+  let mainEvent: EventItem | undefined = undefined;
+  const futureEvents: EventItem[] = [];
+  const currentEvents: EventItem[] = [];
+  const pastEvents: EventItem[] = [];
+  // Sort Events
+  events.sort((a, b) => a.date.getTime() - b.date.getTime());
+  // Process Events
+  for (const event of events) {
+    // Sanity TypeChecking
+    if (event.date < event.open_date)
+      throw new Error('ConfigError: Event opens after event date');
+    // handle main event
+    if (event.main_event) {
+      if (mainEvent == null) mainEvent = event;
+      else throw new Error('ConfigError: Multiple events marked as main');
+    }
+    // Determine if it is future, current or past
+    const now = new Date();
+    if (event.open_date > now) {
+      futureEvents.push(event);
+    } else if (event.date > now) {
+      currentEvents.push(event);
+    } else {
+      pastEvents.push(event);
+    }
+  }
+  // If no main event we will pick a featured event
+  if (mainEvent == null)
+    mainEvent = currentEvents[0] ?? futureEvents[0] ?? pastEvents[-1];
```

Comment:
> why `pastEvents[-1]`. Wont that just be `undefined`

so bassically the behaviour is:
1) Pick main event if exists
2) otherwise pick the next event from `currenEvents`
3) otherwise pick the next event from `futureEvents`
4) If we dont have any upcoming events we display the last event from `pastEvents`.

I do kind of make the assupmtion that we have at least one event here, but its also worth noting that it fails in a pretty good way naturally so I dont feel like we need to handle the edge case that should never happen here.